### PR TITLE
feat: enable working directory

### PR DESCRIPTION
**Pull request by nissim-natanov-outreach**
This PR enables working directory support for all packages.Load call. It is required when running unit tests in isolated directories with their own go.mod files - in this case the packages.Load call must be executed inside that directory.

This PR preserves the original ParseDocs method signature for back compat, introducing new DocsParser to process the comments.

This PR also removes the use of go/importer package. It does not support the new go/modules well and both of its Import and ImportFrom methods are marked as deprecated. The golang.org/x/tools/go/packages is supposed to supersede the go/importer package, and it is already heavily used to parse docs and find convert methods, so there is no reason using the go/importer anymore. Moreover, there is no need to re-parse sources ... so I updated code to store the scope on the Converter itself.

### CODE REVIEW

**Comment by nissim-natanov-outreach** [comment]
@jmattheis Jannis, please take a look at this PR, share your thoughts!

File: generator/generate.go [hunk 1]
```diff
@@ -2,8 +2,6 @@ package generator
 
 import (
 	"fmt"
-	"go/importer"
```
Comment (jmattheis) [inline generator/generate.go:5]: Nice!

File: generator/generate.go [hunk 2]
```diff
@@ -17,6 +15,7 @@ import (
 type Config struct {
 	Name          string
 	ExtendMethods []string
+	WorkingDir    string
```
Comment (jmattheis) [inline generator/generate.go:18]: > This PR enables working directory support for all packages.Load call. It is required when running unit tests in isolated directories with their own go.mod files - in this case the packages.Load call must be executed inside that directory.

Just to make clear, I understand this correctly. You have a repository with multiple go modules inside, and one goverter call should create converters for all these modules? Was it possible before this change to call goverter multiple times inside the directories of your go module?

**Review by jmattheis — CHANGES_REQUESTED** [review]
Hey @nissim-natanov-outreach, thanks for this PR. See sub comments.

File: comments/parse_docs.go [hunk 3]
```diff
@@ -42,10 +48,17 @@ type Method struct {
 
 // ParseDocs parses the docs for the given pattern.
 func ParseDocs(pattern string) ([]Converter, error) {
```
[the remark this answers is not in the document]
Comment (nissim-natanov-outreach) [reply comments/parse_docs.go:52]: I will revert it back to ParseDocs + ParseDocsConfig.

File: generator/generate.go (hunk 1 above)
Comment (nissim-natanov-outreach) [reply generator/generate.go:5]: 👍

File: generator/generate.go (hunk 2 above)
Comment (nissim-natanov-outreach) [reply generator/generate.go:18]: My use case is slightly different: unit tests. Our codegen tool wraps goverter codegen and invokes it as a library (not via code). When our unit test runs, it creates a temporary directory, drops a go.mod with sources and invokes a codegen (and repeats it few times for different test cases). In each test case we need goverter to work in a scope of that temp (unit test) directory, and ignore the current dir.

P.S. I also noticed by removing go/importer usage the tool became a bit faster. I am not sure what was wrong with the go/importer (other than its lack of support for working dirs), but I did notice a big drop in exec time:
* with go/importer and unit test forced to do chdir before goverter: ~15-20s
* with current code: no need to do chdir: ~3-5s

File: comments/parse_docs.go [hunk 4]
```diff
@@ -4,6 +4,7 @@ import (
 	"bufio"
```
[the remark this answers is not in the document]
Comment (nissim-natanov-outreach) [reply comments/parse_docs.go:4]: yea, it does actually, except there is one limitation. Assuming current dir is a module root, with go.mod in it.

OK: go run github.com/jmattheis/goverter/cmd/goverter ./example/simple
Does not work: go run github.com/jmattheis/goverter/cmd/goverter example/simple

E.g. path must start with ./ or ../ to indicate it is a directory - not a package. I will update the readme.

**Review by jmattheis — APPROVED** [review]

**Pull request merged**